code/parachain/frame/assets-transactor-router/src/tests/routing.rs

```diff
@@ -0,0 +1,1092 @@
+				2500,
+			));
+			assert_ok!(<Pallet<Test> as MultiReservableCurrency<AccountId>>::reserve(
+				asset_id_local,
+				&ACCOUNT_LOCAL,
+				900,
+			));
+			assert_ok!(<Pallet<Test> as MultiReservableCurrency<AccountId>>::reserve(
+				asset_id_foreign,
+				&ACCOUNT_FOREIGN,
+				2000,
+			));
+
+			assert_eq!(
+				<Pallet::<Test> as MultiReservableCurrency<AccountId>>::slash_reserved(
+					NATIVE_ASSET_ID,
+					&ACCOUNT_NATIVE,
+					2500
+				),
+				0
+			);
+			assert_eq!(
+				<Pallet::<Test> as MultiReservableCurrency<AccountId>>::slash_reserved(
+					asset_id_local,
+					&ACCOUNT_LOCAL,
+					900
+				),
+				0
+			);
+			assert_eq!(
+				<Pallet::<Test> as MultiReservableCurrency<AccountId>>::slash_reserved(
+					asset_id_foreign,
+					&ACCOUNT_FOREIGN,
+					2000
+				),
+				0
+			);
+
+			assert_eq!(<<Test as Config>::NativeTransactor>::total_balance(&ACCOUNT_NATIVE), 500);
+			assert_eq!(
+				<<Test as Config>::LocalTransactor>::total_balance(asset_id_local, &ACCOUNT_LOCAL),
+				100
+			);
+			assert_eq!(
+				<<Test as Config>::ForeignTransactor>::total_balance(
+					asset_id_foreign,
+					&ACCOUNT_FOREIGN
+				),
+				0
+			);
+			assert_eq!(<<Test as Config>::NativeTransactor>::reserved_balance(&ACCOUNT_NATIVE), 0);
+			assert_eq!(
+				<<Test as Config>::LocalTransactor>::reserved_balance(
+					asset_id_local,
+					&ACCOUNT_LOCAL
+				),
+				0
+			);
+			assert_eq!(
+				<<Test as Config>::ForeignTransactor>::reserved_balance(
+					asset_id_foreign,
+					&ACCOUNT_FOREIGN
+				),
+				0
+			);
+		});
+	}
+}
+
+mod fungibles_route {
+	use frame_support::{
+		assert_ok,
+		traits::{
+			fungibles::{Inspect, InspectHold, Mutate, MutateHold, Transfer, Unbalanced},
+			Currency,
+		},
+	};
+	use orml_traits::MultiCurrency;
+
+	use crate::mocks::AccountId;
+
+	use super::*;
+
+	#[test]
+	fn set_balance() {
+		// set balances for assets via common interface, check that total balance is correct per
+		// pallet
+		new_test_ext().execute_with(|| {
+			let (asset_id_local, asset_id_foreign) = create_assets();
+			assert_ok!(<Pallet<Test> as Unbalanced<AccountId>>::set_balance(
+				NATIVE_ASSET_ID,
+				&ACCOUNT_NATIVE,
+				3000,
+			));
+			assert_ok!(<Pallet<Test> as Unbalanced<AccountId>>::set_balance(
+				asset_id_local,
+				&ACCOUNT_LOCAL,
+				1000,
+			));
+			assert_ok!(<Pallet<Test> as Unbalanced<AccountId>>::set_balance(
+				asset_id_foreign,
+				&ACCOUNT_FOREIGN,
+				2000,
+			));
+
+			assert_eq!(<<Test as Config>::NativeTransactor>::total_balance(&ACCOUNT_NATIVE), 3000);
+			assert_eq!(
+				<<Test as Config>::LocalTransactor>::total_balance(asset_id_local, &ACCOUNT_LOCAL),
+				1000
+			);
+			assert_eq!(
+				<<Test as Config>::ForeignTransactor>::total_balance(
+					asset_id_foreign,
+					&ACCOUNT_FOREIGN
+				),
+				2000
+			);
+		});
+	}
+
+	#[test]
+	fn set_total_issuance() {
+		// set assets' issuance for assets via common interfact, check that total issuance is
+		// correct per pallet
```

Comment:
NIT: you mention "per pallet" in all of these comments, but I think that "per transactor" would make more sense. Honestly, you don't need to mention that part anyways, since that's what these tests are testing anyways. When I suggested adding comments, I meant *within* the tests; such as explaining what the numbers meant. Perhaps this could be resolved better by extracting the numbers to local constants? Something like `const NATIVE_ASSET_TRANSFER_AMOUNT: u128 = 3000;`